src/components/dashboard/dashboardItem/CPUConsumption.js

```diff
@@ -0,0 +1,196 @@
+import ChartDataLabels from "chartjs-plugin-datalabels";
+import { useQuery } from "react-query";
+import {
+    Chart as ChartJS,
+    CategoryScale,
+    LinearScale,
+    BarElement,
+    Title,
+    Tooltip,
+    Legend,
+} from "chart.js";
+import { Bar } from "react-chartjs-2";
+import {
+    getCPUHoursConsumption,
+    CPU_HOURS_CONSUMPTION_QUERY_KEY,
+} from "serviceFacades/dashboard";
+import constants from "../../../constants";
+import ErrorTypographyWithDialog from "components/error/ErrorTypographyWithDialog";
+import { getErrorCode } from "components/error/errorCode";
+import { Typography, useTheme } from "@material-ui/core";
+import { Skeleton } from "@material-ui/lab";
+import { formatDateObject } from "components/utils/DateFormatter";
+
+ChartJS.register(
+    CategoryScale,
+    LinearScale,
+    BarElement,
+    Title,
+    Tooltip,
+    Legend,
+    ChartDataLabels
+);
+
+const options = (usage, quota, timestamp, title, theme, t) => {
+    const formatValue = (value) => `${value} ${t("coreHours")}`;
+    let divisor =
+        quota % 4 === 0
+            ? 4
+            : quota % 3 === 0
+                ? 3
+                : quota % 5 === 0
+                    ? 5
+                    : quota % 2 === 0
+                        ? 2
+                        : 4;
+    let maxTicks = 6;
+    let stepSize =
+        usage <= (quota / divisor) * maxTicks
+            ? Math.ceil(quota / divisor)
+            : quota;
+    return {
+        indexAxis: "y",
+        plugins: {
+            title: {
+                display: true,
+                text: [
+                    title,
+                    t("consumptionChartSecondaryTitle", {
+                        percentage: ((usage / quota) * 100).toFixed(2),
+                        timestamp,
+                    }),
+                ],
+            },
+            legend: {
+                display: false,
+            },
+            tooltip: {
+                callbacks: {
+                    label: (context) => context.dataset.label || "",
+                },
+                xAlign: "right",
+                yAlign: "top",
+            },
+            datalabels: {
+                display: () => (100 * usage) / quota > 20, //if usage > 20% of quota, show data labels
+                backgroundColor: function (context) {
+                    return context.dataset.backgroundColor;
+                },
+                borderRadius: 4,
+                color: theme.palette.primary.contrastText,
+                font: {
+                    weight: "bold",
+                },
+                formatter: (value) => `${t("coreHoursUsed", { value, quota })}`,
+                padding: 1,
+            },
+        },
+        responsive: false,
+        scales: {
+            x: {
+                barThickness: "flex",
+                stacked: false,
+                min: 0,
+                max: Math.max(Math.ceil(usage / stepSize) * stepSize, quota),
```

Comment:
This logic, plus all the logic for calculating `divisor`, `maxTicks`, and `stepSize` is just a copy of the logic from the `DataConsumption` component:
https://github.com/cyverse-de/sonora/blob/1fc3baf1611485cc0e07508668b5793d1f691014/src/components/dashboard/dashboardItem/DataConsumption.js#L48-L63

I think this is complicated enough that it deserves to be refactored into its own util function, which only needs `usage` and `quota` params, and it can return this `Math.max` calculation.

Reply:
Oh wait, I didn't notice before that `DataConsumption` is using `quotaGiB` for `divisor`, but just `quota` for `stepSize` 🤔 

In that case, we can pass on trying to refactor this logic in this PR. We can revisit it later, especially if we have another chart to add in the future that will need this logic as well.

Reply:
Yeah, if we wanted to refactor it out, what we'd probably want to do is:

 * instead of `constants.ONE_GiB`, pass in a unit size as an argument (for CPUs, it might just be 1, but for data it's 1 GiB)
 * calculate divisor using `quota/unitSize`
 * calculate step size using divisor & max ticks

Reply:
I guess if we really wanted to get fancy we could also pass in divisor preferences (list of prefs + default, rather than the implicit `[4, 3, 5, 2], 4` we have here), haha